### PR TITLE
feat: add option to embeed screenshots

### PR DESCRIPTION
**Pull request by leonardosimoura**
Embedded external screenshots into HTML using base64, combined with inlineAssets option produces a single HTML file
Add new option "embeddedScreenshots" in reporterOptions
for #30

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -35,6 +36,65 @@ async function getSimpleCypressConfig(config) {
...
+  for (let i = 0; i < dirResult.length; i++) {
+    const element = dirResult[i];
+    const filePath = path.resolve(jsonsDir,element);
+    jsonParsed.push({
+      file: filePath, data: await fse.readJson(filePath)
+    });
+  }
+
+  function convertContextImgToBase64(contextObj){
+    if (contextObj.value) {
```
Comment (LironEr) [inline lib/utils.js:52]: The user can also add context, so we need to verify that we don't change user context
`context` can also be a simple string without `title` & `value`.
let's validate `context` is an object & the value starts with `screenshots`

```suggestion
if (typeof context === 'object' && context.value && context.title && context.value.startsWith('screenshots')) {```

File: README.md [hunk 2]
```diff
@@ -74,7 +74,8 @@ If you want to customize your HTML report with [mochawesome-report-generator fla
   "reporterOptions": {
     "reportDir": "cypress/report",
     "charts": true,
-    "reportPageTitle": "custom-title"
+    "reportPageTitle": "custom-title",
+    "embeddedScreenshots": true // embedded external screenshots into HTML using base64, use with inlineAssets option to produces a single HTML file
```
Comment (LironEr) [inline README.md:78]: add 
```
"embeddedScreenshots": true,
"inlineAssets": true
```
to `examples/mochawesome-flags/cypress.json`
so the tests would cover the new feature

File: lib/utils.js [hunk 3]
```diff
@@ -35,6 +36,65 @@ async function getSimpleCypressConfig(config) {
...
+
+  for (let i = 0; i < dirResult.length; i++) {
+    const element = dirResult[i];
+    const filePath = path.resolve(jsonsDir,element);
+    jsonParsed.push({
+      file: filePath, data: await fse.readJson(filePath)
+    });
+  }
+
+  function convertContextImgToBase64(contextObj){
```
Comment (LironEr) [inline lib/utils.js:51]: I prefer that `convertContextImgToBase64` & `genereatebase64Context` wouldn't be an inner functions of `convertJsonScreenshotsToBase64`, you can move them outside of the function block

**Review by LironEr — CHANGES_REQUESTED** [review]

**Comment by lacour-vincent** [comment]
Nice idea to add this option to inline screenshots in base64 !

**Comment by leonardosimoura** [comment]
Requested changes made

**Review by LironEr — APPROVED** [review]

**Comment by LironEr** [comment]
Released in [v2.1.0](https://github.com/LironEr/cypress-mochawesome-reporter/releases/tag/v2.1.0)

Thanks @leonardosimoura

**Comment by LironEr** [comment]
Released in [v2.1.0](https://github.com/LironEr/cypress-mochawesome-reporter/releases/tag/v2.1.0)

Thanks @leonardosimoura

**Pull request merged**